src/inmanta/data/__init__.py

```diff
@@ -690,11 +585,14 @@ def as_filter(
         # 1. name of the actual collumn in the DB
         # 2. type of the collumn
         # 3. sanitized value of the collumn
-        filter_elements: List[Tuple[str, ColumnType, object]] = []
 
-        if column_value is not None:
-            # Have column value
-            order_by_collumns_type = self.get_valid_sort_columns()[self.order_by_column]
+        filter_elements: list[tuple[str, ColumnType, object]] = []
+
+        order_by_collumns_type = self.get_order_by_column_type()
+        paging_on_nullable = order_by_collumns_type.nullable and id_value is not None
```

Comment:
What in the case of the two-tuple filtering (id is just the tie-breaker) where we mean to filter on the column value but it happens to be `None`? e.g. consider the following scenario where `(column_value, id_value)` are of type `(Optional[int], int)`:

```
1 1
null 2
2 3
null 4
1 5
null 6
3 7
null 8
```
Suppose we want to filter on `null 6`, this would give `column_value=None, id_value=2`. By the logic implemented here, this would be translated to `table_prefix.id > 6`. We'd get only the last two rows while the expected result would be
```
null 8
1 1
1 5
2 3
3 7
```

Reply:
We always add the second filter when paging_on_nullable, so it would filter on `(column_value, id_value) < (Null, 6)` (with coalesce of course)